csharp/examples/IoTDataPipelineExample/Model/SensorDataPipeline.cs

```diff
@@ -0,0 +1,225 @@
+        private readonly Dictionary<int, TimestampArray.Builder> _colTimestampBuilderDict;
+        private readonly Dictionary<int, DoubleArray.Builder> _colXAxisBuilderDict;
+        private readonly Dictionary<int, DoubleArray.Builder> _colYAxisBuilderDict;
+        private readonly Dictionary<int, DoubleArray.Builder> _colZAxisBuilderDict;
+
+        public Dictionary<string, string> activityLabel = new Dictionary<string, string>()
+            {
+                {"walking", "A"},
+                {"jogging", "B"},
+                {"stairs", "C"},
+                {"sitting", "D"},
+                {"standing", "E"},
+                {"typing", "F"},
+                {"teeth", "G"},
+                {"soup", "H"},
+                {"chips", "I"},
+                {"pasta", "J"},
+                {"drinking", "K"},
+                {"sandwich", "L"},
+                {"kicking", "M"},
+                {"catch", "O"},
+                {"dribbling", "P"},
+                {"writing", "Q"},
+                {"clapping", "R"},
+                {"folding", "S"},
+            };
+
+        public SensorDataPipeline(int totalInputs, int queueCapacity)
+        {
+            _totalInputs = totalInputs;
+            _queueCapacity = queueCapacity;
+            _channel = Channel.CreateBounded<SensorData>(_queueCapacity);
+            _writer = _channel.Writer;
+            _reader = _channel.Reader;
+            _colSubjectIdBuilderDict = new Dictionary<int, Int32Array.Builder>();
+            _colActivityLabelBuilderDict = new Dictionary<int, StringArray.Builder>();
+            _colTimestampBuilderDict = new Dictionary<int, TimestampArray.Builder>();
+            _colXAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+            _colYAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+            _colZAxisBuilderDict = new Dictionary<int, DoubleArray.Builder>();
+        }
+
+        public async Task WriteToChannel(int taskNumber)
+        {
+            Random rand = new Random();
+            List<string> keyList = new List<string>(activityLabel.Keys);
+            int count = keyList.Count;
+            var basis = DateTimeOffset.UtcNow;
+
+            Console.WriteLine($"Write to channel task {taskNumber} started!");
+            while (await _writer.WaitToWriteAsync())
+            {
+                string randomKey = keyList[rand.Next(count)];
+                string label = activityLabel[randomKey];
+
+                // generate random missing values
+                if (rand.Next(10_000) == 9_999)
+                {
+                    label = null;
+                }
+
+                var item = new SensorData
+                {
+                    // approximately 9_000 unique subjects/sensors
+                    subjectId = rand.Next(1_001, 1_123),
+                    activityLabel = label,
+                    timestamp = basis.AddMilliseconds(1),
```

Comment:
Every timestamp in this channel is going to get the same timestamp, right? That seems wrong. Maybe this should just be:

```suggestion
                    timestamp = DateTimeOffset.UtcNow;
```